CMakeLists.txt

```diff
@@ -125,6 +125,11 @@ check_symbol_exists(__get_cpuid cpuid.h HAVE_CPUID)
 check_symbol_exists(strlcpy string.h HAVE_STRLCPY)
 check_symbol_exists(strlcat string.h HAVE_STRLCAT)
 
+check_c_source_compiles("
```

Comment:
Might be worth adding a comment here that on Linux `strerror_r` is switched to the GNU version by `_GNU_SOURCE`, which is defined implicitly in `cmake/os.cmake`

Reply:
Makes sense. Added a comment.